pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -305,7 +324,8 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 					interval = 30
 				} else if numLeftFiles >= 50 && numLeftFiles < 200 {
 					interval = 15
-				} else if numLeftFiles >= 200 && numLeftFiles < 1000 {
+				} else if numLeftFiles >= 200 && numLeftFiles < 1000 || prevBackoff < 7200 {
+					// if recently had received code 429, keep minimum interval as 8 seconds
 					interval = 8
```

Comment:
Shouldn't the block comment (which refers to 3 seconds but not 8) also be updated with this new logic?

And I don't understand why you only check prevBackoff if the number of files exceed 200. Wouldn't it be more logical for example to multiply the interval in all cases with some constant (such as  3) if we've had a backoff in the last two hours?

Reply:
maybe it's good to doubling them too.